holidays/countries/canada.py

```diff
@@ -220,6 +220,14 @@ def _populate(self, year):
         if year >= 1894:
             self[date(year, SEP, 1) + rd(weekday=MO)] = "Labour Day"
 
+        # Funeral of Queen Elizabeth II
+        # https://www.narcity.com/provinces-territories-will-have-a-day-off-monday-mourn-queen
+        # TODO: the territories holiday status (NT, NU, YT) is still tentative
+        queen_funeral_observers = ("BC", "NB", "NL", "NS", "PE", "YT")
+        if self.subdiv in queen_funeral_observers and year == 2022:
+            holiday_name = "Funeral of Her Majesty the Queen Elizabeth II"
+            self[date(year, SEP, 19)] = holiday_name
```

Comment:
A suggestion:

`self[date(year, SEP, 19)] = "Funeral of Her Majesty the Queen Elizabeth II"`

No need to define `holiday_name` here.